concent_api/core/payments/bankster.py

```diff
@@ -71,96 +84,214 @@ def claim_deposit(
+            pk=deposit_claim.payer_deposit_account_id
         )
 
-    # Handle provider claim payment info
-    if settings.ADDITIONAL_VERIFICATION_COST == 0 or concent_use_case != ConcentUseCase.ADDITIONAL_VERIFICATION:
-        providers_claim_payment_info = ClaimPaymentInfo(0, 0)
-    elif available_provider_claim == 0:
-        providers_claim_payment_info = ClaimPaymentInfo(0, settings.ADDITIONAL_VERIFICATION_COST)
-    else:
-        transaction_hash = service.cover_additional_verification_cost(  # pylint: disable=no-value-for-parameter
-            provider_eth_address=provider_ethereum_address,
-            value=available_provider_claim,
-            subtask_id=subtask_id,
-        )
-        providers_claim_payment_info = ClaimPaymentInfo(
-            amount_paid=available_provider_claim,
-            amount_pending=settings.ADDITIONAL_VERIFICATION_COST - available_provider_claim,
-            tx_hash=transaction_hash,
-            payment_ts=current_time,
+        # Bankster sums the amounts of all existing DepositClaims that have the same payer as the one being processed.
+        aggregated_client_claims = DepositClaim.objects.filter(
+            payer_deposit_account=deposit_claim.payer_deposit_account
+        ).exclude(
+            pk=deposit_claim.pk
+        ).aggregate(
+            sum_of_existing_claims=Coalesce(Sum('amount'), 0)
         )
 
-    return requestors_claim_payment_info, providers_claim_payment_info
+        # Bankster subtracts that value from the amount of funds available in the deposit.
+        available_funds_without_claims = available_funds - aggregated_client_claims['sum_of_existing_claims']
+
+        # If the result is negative or zero, Bankster removes the DepositClaim object being processed.
+        if available_funds_without_claims <= 0:
+            deposit_claim.delete()
+            return None
+
+        # Otherwise if the result is lower than DepositAccount.amount,
+        # Bankster sets this field to the amount that's actually available.
+        elif available_funds_without_claims < deposit_claim.amount:
+            deposit_claim.amount = available_funds_without_claims
+
+        # If the DepositClaim still exists at this point, Bankster uses SCI to create an Ethereum transaction.
+        if deposit_claim.concent_use_case == ConcentUseCase.FORCED_ACCEPTANCE:
+            ethereum_transaction = service.force_subtask_payment(  # pylint: disable=no-value-for-parameter
+                requestor_eth_address=deposit_claim.payer_deposit_account.ethereum_address,
+                provider_eth_address=deposit_claim.payee_ethereum_address,
+                value=deposit_claim.amount,
+                subtask_id=deposit_claim.subtask_id,
+            )
+        elif deposit_claim.concent_use_case == ConcentUseCase.ADDITIONAL_VERIFICATION:
+            subtask = Subtask.objects.filter(subtask_id=deposit_claim.subtask_id).first()  # pylint: disable=no-member
```

Comment:
Why not `get_one_or_none()`?

Reply:
There is a problem with circular import. And this code does actually the same.

Reply:
Hmm... That's a problem with our module organization. I don't think that `subtask_helpers` is the right module for a very generic function like `get_one_or_none()`. And `_update_timed_out_subtask()` does not seem to be a "helper" either. It's a critical part of our business logic and introduces a dependency on bankster which I wouldn't really expect in a simple helper.

Anyway, the alternative works so let's leave it as is for now. I only asked because using `first()` is always suspicious in database-related code.